cmake/TorchConfig.cmake.in

```diff
@@ -44,8 +44,10 @@ else()
   set(TORCH_LIBRARIES torch)
 endif()
 
-find_library(C10_LIBRARY c10 PATHS "${TORCH_INSTALL_PREFIX}/lib")
-list(APPEND TORCH_LIBRARIES ${C10_LIBRARY})
+if(@BUILD_SHARED_LIBS@)
+  find_library(C10_LIBRARY c10 PATHS "${TORCH_INSTALL_PREFIX}/lib")
+  list(APPEND TORCH_LIBRARIES ${C10_LIBRARY})
+endif()
```

Comment:
Why do you need to keep this rule here conditionally rather than just to move it to line 59? Or are you saying that library link order should be different for shared vs static libs?